services/telemetry/telemetry_test.go

```diff
@@ -17,17 +17,115 @@
 package telemetry
 
 import (
+	"context"
+	"encoding/hex"
+	"os"
 	"testing"
+	"time"
 
+	"github.com/golang/protobuf/proto"
+	"github.com/google/uuid"
+	pmmv1 "github.com/percona-platform/saas/gen/telemetry/events/pmm"
+	reporterv1 "github.com/percona-platform/saas/gen/telemetry/reporter"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
-func TestMakePayload(t *testing.T) {
-	s := NewService(nil, "1.3.1")
-	expected := "ECAB81E4C47D456CA9EC20AEBF91AB44;OS;\nECAB81E4C47D456CA9EC20AEBF91AB44;PMM;1.3.1\n"
-	assert.Equal(t, expected, string(s.makePayload("ECAB81E4C47D456CA9EC20AEBF91AB44"))) // \n are important
+const devTelemetryHost = "callhome-staging.percona.com:443"
+
+//nolint:lll
+func TestMakeV1Payload(t *testing.T) {
```

Comment:
We should figure out how to differentiate pmm v1 - v2 and telemetry v1 - v2

Reply:
I think "v1" and "v2" are good enough for now – hopefully, we will remove v1 soon.